### PR TITLE
Code Review - Faster Updates and Undo option

### PR DESCRIPTION
**Pull request by xhlulu**
By using S3 to store user uploaded images, and redis cache to memoize an stack of actions (operations) applied on the image, the speed was significantly increased, as well as opened up the option to undo an action.

An action is a dictionary of zone selected, type of operation, and the operation itself.

Please view this link **for the dev version**: https://dash-image-processing-dev.herokuapp.com

Please view this one for the current master version (OLD): http://dash-image-processing.herokuapp.com

If you want to test this locally, please let me know and I will send you the .env file

### CODE REVIEW

**Comment by xhlulu** [comment]
@shammamah @matthewchan15 any update?

File: dash_reusable_components.py [hunk 1]
```diff
@@ -218,9 +220,17 @@ def NamedInlineRadioItems(name, short, options, val, **kwargs):
...
+                        enc_format='png',
+                        display_mode='fixed',
+                        dragmode='select',
+                        verbose=False,
+                        **kwargs):
     if enc_format == 'jpeg':
-        encoded_image = pil_to_b64(image, enc_format=enc_format, verbose=verbose, quality=95)
+        if image.mode == 'RGBA':
+            image = image.convert('RGB')
+        encoded_image = pil_to_b64(image, enc_format=enc_format, verbose=verbose, quality=80)
```
Comment (shammamah) [inline dash_reusable_components.py]: I think this line is too long for PEP8; try putting each argument on its own line, like you did for `InteractiveImagePIL`  above

File: utils.py [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
+    'action_stack': []
+})
 
-GRAPH_PLACEHOLDER = drc.InteractiveImagePIL(
-    image_id='interactive-image',
-    image=Image.open('images/default.jpg'),
-    enc_format='png',
-    display_mode='fixed'
-)
+IMAGE_STRING_PLACEHOLDER = drc.pil_to_b64(Image.open('images/default.jpg').copy(), enc_format='jpeg')
```
Comment (shammamah) [inline utils.py]: This line is also too long!

**Comment by xhlulu** [comment]
Ok, made the code length PEP8 conform for everything!

**Comment by xhlulu** [comment]
@shammamah

**Comment by xhlulu** [comment]
@shammamah Let me know if there's anything else to update; I'll try to add my new apps to the gallery on monday.

**Pull request merged**